vizro-core/src/vizro/models/_base.py

```diff
@@ -136,7 +136,11 @@ def _extract_captured_callable_data_info() -> set[str]:
 
 def _add_type_to_union(union: type[Any], new_type: type[Any]):  # TODO[mypy]: not sure how to type the return type
     args = get_args(union)
-    return Union[args + (new_type,)]  # noqa: RUF005 #as long as we support Python 3.9, we can't use the new syntax
+    all_types = args + (new_type,)  # noqa: RUF005 #as long as we support Python 3.9, we can't use the new syntax
+    # The below removes duplicates by type, which would trigger a pydantic error otherwise
+    # Last added type will be the one that is kept - this is replicating V1 behavior that would other raise an error in V2
+    unique_types = tuple({t.model_fields["type"].default: t for t in all_types}.values())
```

Comment:
This took me a which to understand... Maybe worth another comment explaining a bit how it works, like what `t.model_fields["type"].default` contains and how `values()` selects the last value if a key is duplicated.